cassiopeia/ProcessingPipeline/process/alignment_utilities.py

```diff
@@ -0,0 +1,214 @@
+"""
+This file stores useful functions for dealing with alignments.
+"""
+import re
+from typing import List, Tuple
+
+
+class UnknownCigarStringError(Exception):
+    pass
+
+
+def parse_cigar(
+    cigar: str,
+    seq: str,
+    ref: str,
+    start_pos: int,
+    barcode_interval: Tuple[int, int],
+    cutsites: List[int],
+    cutsite_window: int = 0,
+    context: bool = True,
+    context_size: int = 5,
+) -> Tuple[str, List[str]]:
+    """Parse the cigar string from a TargetSite alignment.
+
+    Parse the cigar string from an alignment of a TargetSite read into the
+    indels observed at each of the cutsites. We assume that the construct
+    has at least an integration barcode marking reads from the same Target Site.
+    It's also likely that there is at least one cut site per Target Site, whose
+    location can be specified with the cutsites list, though if this is an
+    empty list, no cut sites will be processed. After aligning, the intBC and a
+    list storing the indels at each cutsite is returned.
+
+    Args:
+        cigar: CIGAR string from the alignment
+        seq: Query sequence
+        ref: Reference sequence
+        barcode_interval: Interval in refernece seqeunce that stores the
+            integration barcode
+        cutsites: A list of cutsite locations in the reference
+        cutsite_window: Number of nucleotides to the left and right of the
+            cutsite to look for the beginning of an indel
+        context: Include nuceotide sequence around indels
+        context_size: Number of bases to report for context
+
+    Returns:
+        The intBC and a list storing the indel observed at each cutsite.
+    """
+
+    cutsite_lims = [
+        (site - cutsite_window, site + cutsite_window) for site in cutsites
+    ]
+    indels = ["" for site in cutsites]
+    uncut_indels = ["" for site in cutsites]
+    intBC = ""
+
+    ref_anchor = ref[barcode_interval[0] - 11 : barcode_interval[0]]
+    intBC_length = barcode_interval[1] - barcode_interval[0]
+
+    # set up two pointers to the reference and query, respectively
+    ref_pointer = start_pos
+    query_pointer = 0
```

Comment:
I think we should initialize query_pointer to query_begin found in the documentation for skbio's AlignmentStructure:
http://scikit-bio.org/docs/0.5.1/generated/skbio.alignment.AlignmentStructure.html#skbio.alignment.AlignmentStructure
and also the ref_pointer to target_begin. These can be passed in from call_indels